### PR TITLE
Add bob.core recipe

### PR DESCRIPTION
**Pull request by 183amir**

### CODE REVIEW

**Comment by 183amir** [comment]
This was working on [Linux](https://travis-ci.org/bioidiap/bob.conda/jobs/114218332) before. I do not know what has changed. This was the [recipe](https://github.com/bioidiap/bob.conda/blob/master/bob.core/meta.yaml). There is no code difference between the versions, only version got bumped.

**Comment by 183amir** [comment]
Hey guys can I add something like this to `bob` recipes:
```
  script: CXXFLAGS="$CXXFLAGS -stdlib=libc++ " python -B setup.py install --single-version-externally-managed --record record.txt  # [osx]
  script: python -B setup.py install --single-version-externally-managed --record record.txt
```
`bob` packages use `boost` heavily and `CXXFLAGS="$CXXFLAGS -stdlib=libc++ "` should be added to our `mac` builds.

**Comment by jakirkham** [comment]
> `bob` packages use `boost` heavily and `CXXFLAGS="$CXXFLAGS -stdlib=libc++ "` should be added to our `mac` builds.

Agreed. Also, may want to set C++11 support if you need it. We build Boost like [this]( https://github.com/conda-forge/boost-feedstock/blob/master/recipe/build.sh#L17-L21 ).

> Hey guys can I add something like this to `bob` recipes:
```
  script: CXXFLAGS="$CXXFLAGS -stdlib=libc++ " python -B setup.py install --single-version-externally-managed --record record.txt  # [osx]
  script: python -B setup.py install --single-version-externally-managed --record record.txt
```

If you are having to set environment variables, personally I would prefer to just see `build.sh`. It will be cleaner, easier to read, and if you find you need more stuff like this it will be easier to add.

**Comment by 183amir** [comment]
That's why I was arguing not removing `build.sh` back in [here](https://github.com/conda-forge/staged-recipes/pull/264#discussion_r58353082) :D

**Comment by jakirkham** [comment]
That case is different as we don't need to anything, but a simple one liner. Putting it all in the same document makes it easier to follow. The change here makes that one line harder to follow so it is better off in its own document.

**Comment by 183amir** [comment]
Yes but I have to add these flags there too. But honestly I like this one liner better because of `[osx]` selector.

**Comment by 183amir** [comment]
@jakirkham do you know how can I prevent travis from adding `-mmacosx-version-min=10.6 ` to the [build command](https://travis-ci.org/conda-forge/staged-recipes/jobs/121461324#L484)?

File: recipes/bob.core/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+source:
+  fn: bob.core-{{ version }}.zip
+  url: https://pypi.python.org/packages/source/b/bob.core/bob.core-{{ version }}.zip
+  md5: 26c12e1384944ff3873b08409609d151
+
+build:
+  number: 0
+  skip: true  # [win]
+  script: python -B setup.py install --single-version-externally-managed --record record.txt
+  script: LINKFLAGS="$LINKFLAGS -stdlib=libc++ -mmacosx-version-min=10.7 " CXXFLAGS="$CXXFLAGS -stdlib=libc++ -mmacosx-version-min=10.7 " python -B setup.py install --single-version-externally-managed --record record.txt  # [osx]
```
Comment (jakirkham) [inline recipes/bob.core/meta.yaml]: Please break this out into a separate `build.sh`.

Comment (183amir) [inline recipes/bob.core/meta.yaml]: Once it works, I will.

Comment (jakirkham) [inline recipes/bob.core/meta.yaml]: I'm sorry, but it is hard to read and you are asking me to help you figure out what is wrong. While I'm happy to help, this is just a bit painful to read this long one line string. Please break it out before we continue.

Comment (ocefpaf) [inline recipes/bob.core/meta.yaml]: @183amir we prefer the `build/script` field in the `meta.yaml` when we have simple `python setup.py ...`, but we strongly recommend a `build.sh` when it gets complex.  Please re-consider your choice here.

**Comment by 183amir** [comment]
Still cannot figure this out.

File: recipes/bob.core/build.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+if [[ `uname` == 'Darwin' ]]; then
+	MACOSX_VERSION_MIN=10.7
```
Comment (jakirkham) [inline recipes/bob.core/build.sh]: Try setting `CMAKE_OSX_DEPLOYMENT_TARGET=MACOSX_VERSION_MIN`. Also, try adding `export` in front of all the `bash` variables and let's see how that goes.

File: recipes/bob.core/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/bob.core/meta.yaml]: Thank you for breaking this out.

**Comment by 183amir** [comment]
Ok guys, since I do not own a mac machine and I am not getting any support from my colleagues for this work and we only use our software on Linux, I am skipping the mac builds. However, I will keep the mac specific build instructions hoping that someone builds on top of it.
So, if you do not have any objection, please merge this.

**Comment by ocefpaf** [comment]
Open an issue at the feedstock regarding the OS X build and let's get this merged!

**Comment by ocefpaf** [comment]
It looks like you got it working though.

**Comment by 183amir** [comment]
Yes, @ocefpaf. I tried on more time and it worked.

**Pull request merged**